ci/build-test-matrix.js

```diff
@@ -5,154 +5,241 @@
-//   `default`
-const array = [
-  {
-    "os": "ubuntu-latest",
-    "name": "Test Linux x86_64",
-    "filter": "linux-x64",
-    "isa": "x64",
-    "extra_features": "--features wasmtime-wasi-nn/onnx"
-  },
-  {
-    "os": "ubuntu-latest",
-    "name": "Test MSRV on Linux x86_64",
-    "filter": "linux-x64",
-    "isa": "x64",
-    "rust": "msrv",
-  },
-  {
-    "os": "ubuntu-latest",
-    "name": "Test Linux x86_64 with MPK",
-    "filter": "linux-x64",
-    "isa": "x64"
-  },
-  {
-    "os": "macos-13",
-    "name": "Test macOS x86_64",
-    "filter": "macos-x64",
-    "extra_features": "--features wasmtime-wasi-nn/onnx"
-  },
-  {
-    "os": "macos-14",
-    "name": "Test macOS arm64",
-    "filter": "macos-arm64",
-    "target": "aarch64-apple-darwin",
-    "extra_features": "--features wasmtime-wasi-nn/onnx"
-  },
-  {
-    "os": "windows-latest",
-    "name": "Test Windows MSVC x86_64",
-    "filter": "windows-x64",
-    "extra_features": "--features wasmtime-wasi-nn/onnx"
-  },
-  {
-    "os": "windows-latest",
-    "target": "x86_64-pc-windows-gnu",
-    "name": "Test Windows MinGW x86_64",
-    "filter": "mingw-x64"
-  },
-  {
-    "os": "ubuntu-latest",
-    "target": "aarch64-unknown-linux-gnu",
-    "gcc_package": "gcc-aarch64-linux-gnu",
-    "gcc": "aarch64-linux-gnu-gcc",
-    "qemu": "qemu-aarch64 -L /usr/aarch64-linux-gnu",
-    "qemu_target": "aarch64-linux-user",
-    "name": "Test Linux arm64",
-    "filter": "linux-arm64",
-    "isa": "aarch64",
-  },
-  {
-    "os": "ubuntu-latest",
-    "target": "s390x-unknown-linux-gnu",
-    "gcc_package": "gcc-s390x-linux-gnu",
-    "gcc": "s390x-linux-gnu-gcc",
-    "qemu": "qemu-s390x -L /usr/s390x-linux-gnu",
-    "qemu_target": "s390x-linux-user",
-    "name": "Test Linux s390x",
-    "filter": "linux-s390x",
-    "isa": "s390x"
-  },
-  {
-    "os": "ubuntu-latest",
-    "target": "riscv64gc-unknown-linux-gnu",
-    "gcc_package": "gcc-riscv64-linux-gnu",
-    "gcc": "riscv64-linux-gnu-gcc",
-    "qemu": "qemu-riscv64 -cpu rv64,v=true,vlen=256,vext_spec=v1.0,Zfa=true,zba=true,zbb=true,zbc=true,zbs=true,zbkb=true,zcb=true -L /usr/riscv64-linux-gnu",
-    "qemu_target": "riscv64-linux-user",
-    "name": "Test Linux riscv64",
-    "filter": "linux-riscv64",
-    "isa": "riscv64",
-  }
-];
-
-for (let config of array) {
-  if (config.rust === undefined) {
-    config.rust = 'default';
-  }
+async function getWorkspaceMembers() {
+  const child = spawn("cargo", ["metadata"], { encoding: "utf8" });
+  let data = "";
+  child.stdout.on("data", chunk => data += chunk);
+  await new Promise((resolve, reject) => {
+    child.on("close", resolve);
+    child.on("error", reject);
+  });
+  const metadata = JSON.parse(data);
+  return metadata.workspace_members.map(m => {
+    const matches = m.match(/#(.*)@\d+\.\d+\.\d+$/)
+          ||  m.match(/([^/]*)#\d+\.\d+\.\d+$/);
+    if (!matches) throw new Error("failed to parse crate name from: " + m);
+    return matches[1];
```

Comment:
Technically-speaking rather than using a regex for this it's probably best to build a map based on the output of `cargo metadata` and lookup things bad on id to project out the package name from the package metadata in the json blob. (probably more robust than regex-matching)